### PR TITLE
Update paste-gpg-key-id.md

### PR DESCRIPTION
**Pull request by dustinkirkland**
The documentation in here definitively does not work -- adding a trailing exclamation mark on a subkey causes git signing to fail:

```
$ GIT_TRACE=1 git commit -S -a
14:39:24.442514 git.c:455               trace: built-in: git commit -S -a
hint: Waiting for your editor to close the file... 14:39:24.461555 run-command.c:668       trace: run_command: GIT_INDEX_FILE=foo/.git/COMMIT_EDITMSG
14:39:39.240473 run-command.c:668       trace: run_command: gpg --status-fd=2 -bsau 'AD962CEE23420C1C'\!''
error: gpg failed to sign the data
fatal: failed to write commit object

```
However, omitting the trailing "!" works perfectly....

```
$ git config --global user.signingkey "AD962CEE23420C1C"
$ GIT_TRACE=1 git commit -S -a
14:42:37.225614 git.c:455               trace: built-in: git commit -S -a
hint: Waiting for your editor to close the file... 14:42:37.242806 run-command.c:668       trace: run_command: GIT_INDEX_FILE=foo/.git/COMMIT_EDITMSG
14:42:48.298643 run-command.c:668       trace: run_command: gpg --status-fd=2 -bsau AD962CEE23420C1C
14:42:48.601145 run-command.c:668       trace: run_command: git maintenance run --auto --no-quiet
14:42:48.602894 git.c:455               trace: built-in: git maintenance run --auto --no-quiet
[Makefile c850bc3a] ensure that we have a packages directory before mounting it
 1 file changed, 1 insertion(+)

```

### CODE REVIEW

**Comment by nguyenalex836** [comment]
@dustinkirkland Thanks so much for opening a PR! I'll get this triaged for review ✨

**Review by felicitymay — COMMENTED** [review]
Based on the official git documentation, it seems as if the `!` suffix is not needed. See:

- [Git - Signing Your Work](https://git-scm.com/book/en/v2/Git-Tools-Signing-Your-Work)
- [Git - Git Configuration](https://git-scm.com/book/en/v2/Customizing-Git-Git-Configuration)
- [Git - git-config Documentation](https://git-scm.com/docs/git-config)

However, it seems as if there are times where `!` (or escaped `\!`) is required to force using a specific key. See:

- `!` https://www.gnupg.org/documentation/manuals/gnupg/Specify-a-User-ID.html
   > When using gpg an exclamation mark (!) may be appended to force using the specified primary or secondary key and not to try and calculate which primary or secondary key to use.
- `!`https://vitalii.kozlovskyi.dev/posts/gpg-hardware-key-and-git-signing/
- `\!` https://www.hjdskes.nl/blog/psa-github-gpg/

Consequently, I'm going to expand your change to add this as an option.

File: data/reusables/gpg/paste-gpg-key-id.md [hunk 1]
```diff
@@ -4,8 +4,8 @@
    git config --global user.signingkey 3AA5C34371567BD2
    ```
 
-   Alternatively, when setting a subkey include the `!` suffix. In this example, the GPG subkey ID is `4BB6D45482678BE3`:
+   Alternatively, you may want to use a subkey. In this example, the GPG subkey ID is `4BB6D45482678BE3`:
 
    ```shell
-   git config --global user.signingkey 4BB6D45482678BE3!
+   git config --global user.signingkey 4BB6D45482678BE3
    ```
```
Comment (felicitymay) [inline data/reusables/gpg/paste-gpg-key-id.md:11]: Updating to cover forcing use of a specific key or subkey:

````suggestion
   git config --global user.signingkey 4BB6D45482678BE3
   ```

If you use multiple keys and subkeys, then you should append an exclamation mark `!` to the key to tell git that this is your preferred key. Sometimes you may need to escape the exclamation mark with a back slash: `\!`.

````

File: data/reusables/gpg/paste-gpg-key-id.md [hunk 2]
```diff
@@ -4,8 +4,11 @@
...
+
+   If you use multiple keys and subkeys, then you should append an exclamation mark `!` to the key to tell git that this is your preferred key. Sometimes you may need to escape the exclamation mark with a back slash: `\!`.
+
```
Comment (felicitymay) [inline data/reusables/gpg/paste-gpg-key-id.md:14]: It looks as if I need to remove the blank line here for the linter:

```suggestion
   If you use multiple keys and subkeys, then you should append an exclamation mark `!` to the key to tell git that this is your preferred key. Sometimes you may need to escape the exclamation mark with a back slash: `\!`.
```

**Comment by felicitymay** [comment]
I'm not sure why the preview's failing - I've tried re-running the workflow, which didn't make any difference 😞

I'm going to update this branch from `main` which will trigger the preview from scratch 🤞🏻

**Comment by felicitymay** [comment]
@dustinkirkland - this has the change you suggested, but still allows people to discover the use of an `!`, should they need it.

Since you're based in the US and I'm in Europe, I'm going to leave this for you to take a look at the preview: [Telling Git about your GPG key](https://docs-33799-b2f5cf.preview.ghdocs.com/en/authentication/managing-commit-signature-verification/telling-git-about-your-signing-key#telling-git-about-your-gpg-key) with the intention of merging it tomorrow unless you spot any errors or concerns.

**Review by felicitymay — APPROVED** [review]

**Pull request merged**